Update ticket. (parte do Issue #93)

robertatakenaka opened the pull request:
Não sei fazer teste para quando o commit não dá certo. Sugestões?

robertatakenaka committed:
Update ticket. (parte do Issue #93)

jamilatta commented on balaio/gateway_server.py:
```diff
@@ -125,6 +125,25 @@ def list_ticket(request):
+    #http://docs.sqlalchemy.org/en/latest/orm/session.html#embedding-sql-insert-update-expressions-into-a-flush
+    ticket = request.db.query(models.Ticket).get(request.matchdict['id'])
+    if ticket:
+        ticket.update(request.PATCH)
```
Mesma erro de vazamento para a camada do modelo.

robertatakenaka committed:
refactoring Ticket.update()

robertatakenaka commented:
Por favor, reavaliar.

robertatakenaka committed:
Removed Ticket.update(). The corresponding code is in the view: update_ticket.

gustavofonseca commented on balaio/models.py:
```diff
@@ -144,6 +152,14 @@ def __init__(self, *args, **kwargs):
         self.started_at = datetime.datetime.now()
         self.is_open = True
 
+    def update(self, is_open, comment_author='', message=''):
```
Também acho que, da mesma maneira que o método `new`, estes códigos cabem bem na view-function.

No máximo eu acho que vale encapsular a ação de _fechar_ o ticket, para ficar com uma interface mais esperta...tipo:

``` python
def close(self):
    self.is_open = False
```

robertatakenaka committed:
Merge branch 'master' of git://github.com/scieloorg/balaio into 93_ticket_update

gustavofonseca commented on balaio/gateway_server.py:
```diff
@@ -125,6 +125,26 @@ def list_ticket(request):
+    Update a ticket
+    """
+    ticket = request.db.query(models.Ticket).get(request.matchdict['id'])
+    if ticket:
```
Se o método `Ticket.close` for criado, aqui poderia ter:

``` python
if request.PATCH['is_open'] == False: 
    ticket.close()
```

robertatakenaka committed:
Fixes test name
Insert fields author and title in Ticket, and nullable=False
Insert field author, and nullable=False

robertatakenaka committed:
Merge branch 'master' of git://github.com/scieloorg/balaio into 93_ticket_update

robertatakenaka commented:
ajustado
